### PR TITLE
Fix missing test module results

### PR DESCRIPTION
**Pull request by Martchus**
* Fix regression introduced by
  https://github.com/os-autoinst/openQA/pull/3849
* See https://progress.opensuse.org/issues/91461

### CODE REVIEW

**Review by okurz — APPROVED** [review]
Too hard to fully grasp for me but have not found anything wrong :)

**Comment by Martchus** [comment]
A unit test is failing but that's actually good because it gives me a starting point for improving the unit tests which should be done anyways.

**Comment by okurz** [comment]
ok, I see that you included a reference to poo but for convenience I suggest in git commit messages to use full URLs, sorry about that.

About the test failures, https://github.com/os-autoinst/os-autoinst/pull/1652/checks?check_run_id=2399230336#step:3:539 can not be due to your changes. I recorded that failure in https://progress.opensuse.org/issues/91491 and when you push another update to the commit the test should retrigger. https://github.com/os-autoinst/os-autoinst/pull/1652/checks?check_run_id=2399230248#step:3:16 fails due to an update in the Dockerfile syntax checker. I will try to take care in some minutes.

**Comment by okurz** [comment]
Intended fix for Dockerfile style issue: https://github.com/os-autoinst/os-autoinst/pull/1653

**Comment by Martchus** [comment]
The commit message only contains full URLs, see https://github.com/os-autoinst/openQA/commit/6e089b14c65f2d2239f69665adbccf9ab406ed37.patch for the raw commit message. There are also no failed tests. Somehow I think you've wrote these comments under the wrong tab.

Btw, I'm currently still doing local testing but so far all test runs look good.

**Comment by okurz** [comment]
Yeah I have written my comments on the completely wrong PR :D

**Comment by okurz** [comment]
As mkittler did not want to merge the revert we have to expedite the merge, hence merging

File: t/24-worker-jobs.t [hunk 1]
```diff
@@ -1228,16 +1228,34 @@ subtest 'optipng' => sub {
...
+      'my_result removed from test order but not further test modules';
+
+    $module_res{$_} = {name => $_} for qw(your_result our_result);
+    ($ret, $last_test_module) = $job->_read_result_file('your_result', $extra_test_order);
+    is $last_test_module, 'your_result', 'last test module is now your_result because that is the one to upload up to';
+
+    $job->_reduce_test_order('our_result');
+    is_deeply $job->test_order, [{name => 'our_result'}], 'our_result preserved as it is still the current module';
+    # note: When pausing on a failed assert/check screen we upload up to the current module so the last module we have
+    #       uploded results for might still be in progress and still needs to be considered on further uploads.
```
Comment (okurz) [inline t/24-worker-jobs.t:1258]: typo, oh no!

**Pull request merged**